Source/JavaScriptCore/jsc.cpp

```diff
@@ -4049,6 +4053,27 @@ void CommandLine::parseArguments(int argc, char** argv)
             m_dumpSamplingProfilerData = true;
             continue;
         }
+#if PLATFORM(COCOA)
+        if (!strcmp(arg, "--crash-vm=PGMOOBLowerGuardPage")) {
+            WTF::forceEnablePGM();
+            char* result = static_cast<char*>(fastMalloc(10000000));
+            result = result - getpagesize();
+            *result = 'a';
+        }
+        if (!strcmp(arg, "--crash-vm=PGMOOBUpperGuardPage")) {
+            WTF::forceEnablePGM();
+            size_t allocSize = getpagesize() * 10000;
+            char* result = static_cast<char*>(fastMalloc(10000000));
+            result = result + allocSize;
+            *result = 'a';
```

Comment:
Let's extract this into a `static NEVER_INLINE` function so that it gets a crash signature like "`jsc:  test_pgm_oob_overflow`":

```
static NEVER_INLINE NO_RETURN_DUE_TO_CRASH void test_pgm_oob_overflow()
{
    WTF::forceEnablePGM();
    size_t allocSize = getpagesize() * 10000;
    char* result = static_cast<char*>(fastMalloc(10000000));
    result = result + allocSize;
    *result = 'a';
    RELEASE_ASSERT_NOT_REACHED();
}
```